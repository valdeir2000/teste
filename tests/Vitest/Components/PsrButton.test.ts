import { mount } from '@vue/test-utils';
import { describe,expect, it } from 'vitest';
import PsrButton from '../../../resources/js/Components/PsrButton.vue';

describe('Testa a renderização do componente', () => {
  const selectorButton = '[data-testid="button"]';

  it('O botão sem tipo e sem color não deve possuir classes adicionais', () => {
    const wrapper = mount(PsrButton);

    expect(wrapper.find(selectorButton).classes().length).toEqual(1);
    expect(wrapper.find(selectorButton).classes()).toContainEqual('button');
  });

  it('O botão do tipo ícone deve possuir a classe icon', () => {
    const wrapper = mount(PsrButton, {
      props: {
        type: 'icon',
      },
    });

    expect(wrapper.find(selectorButton).classes()).toContain('icon');
  });

  it('O botão da cor danger deve possuir a classe danger', () => {
    const wrapper = mount(PsrButton, {
      props: {
        color: 'danger',
      },
    });

    expect(wrapper.find(selectorButton).classes()).toContain('danger');
  });

  it('Ao clicar no botão, ele deve disparar o evento click', async () => {
    const wrapper = mount(PsrButton);

    await wrapper.get(selectorButton).trigger('click');

    expect(wrapper.emitted()).toHaveProperty('click');
  });
});

describe('Verifica snapshot do componente', () => {
  it('Botão padrão', () => {
    const wrapper = mount(PsrButton);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Botão do tipo ícone', () => {
    const wrapper = mount(PsrButton, { props: { type: 'icon' } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Botão da cor danger', () => {
    const wrapper = mount(PsrButton, { props: { color: 'danger' } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
