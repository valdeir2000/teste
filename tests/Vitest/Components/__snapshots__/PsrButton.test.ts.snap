// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Verifica snapshot do componente > Botão da cor danger 1`] = `"<a data-v-9ab42fc5=\\"\\" data-testid=\\"button\\" class=\\"button  danger\\"></a>"`;

exports[`Verifica snapshot do componente > Botão do tipo ícone 1`] = `"<a data-v-9ab42fc5=\\"\\" data-testid=\\"button\\" class=\\"button icon\\"></a>"`;

exports[`Verifica snapshot do componente > Botão padrão 1`] = `"<a data-v-9ab42fc5=\\"\\" data-testid=\\"button\\" class=\\"button\\"></a>"`;
